src/core/tiledmesh/qgscesiumtilesdataprovider.cpp

```diff
@@ -20,45 +20,190 @@
 #include "qgsapplication.h"
 #include "qgsprovidersublayerdetails.h"
 #include "qgsthreadingutils.h"
+#include "qgsnetworkaccessmanager.h"
+#include "qgsblockingnetworkrequest.h"
+#include "qgscesiumutils.h"
 
 #include <QUrl>
 #include <QIcon>
+#include <QNetworkRequest>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QFileInfo>
 
 ///@cond PRIVATE
 
 #define PROVIDER_KEY QStringLiteral( "cesiumtiles" )
 #define PROVIDER_DESCRIPTION QStringLiteral( "Cesium 3D Tiles data provider" )
 
+//
+// QgsCesiumTilesDataProviderSharedData
+//
+
+QgsCesiumTilesDataProviderSharedData::QgsCesiumTilesDataProviderSharedData() = default;
+
+void QgsCesiumTilesDataProviderSharedData::setTilesetContent( const QVariantMap &tileset )
+{
+  mTileset = tileset;
+  mCrs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4979" ) );
+
+  // parse root
+  {
+    const QVariantMap root = tileset.value( QStringLiteral( "root" ) ).toMap();
+    // parse root bounding volume
+    {
+      const QVariantMap rootBoundingVolume = root.value( QStringLiteral( "boundingVolume" ) ).toMap();
+      if ( rootBoundingVolume.contains( QStringLiteral( "region" ) ) )
+      {
+        const QgsBox3d rootRegion = QgsCesiumUtils::parseRegion( rootBoundingVolume.value( QStringLiteral( "region" ) ).toList() );
+        if ( !rootRegion.isNull() )
+        {
+          mZRange = QgsDoubleRange( rootRegion.zMinimum(), rootRegion.zMaximum() );
+          // The latitude and longitude values are given in radians!
+
+          const double xMin = rootRegion.xMinimum() * 180 / M_PI;
+          const double xMax = rootRegion.xMaximum() * 180 / M_PI;
+          const double yMin = rootRegion.yMinimum() * 180 / M_PI;
+          const double yMax = rootRegion.yMaximum() * 180 / M_PI;
```

Comment:
How about having the conversion from radians in QgsCesiumUtils::parseRegion() ?

Reply:
because I'm not confident that the specifications are correct (or will always be followed!). The specs also aren't clear on how the region interacts with the srs metadata for a source... eg will the region ALWAYS be in EPSG:4979, regardless of what the srs metadata states? The docs imply so, but I strongly suspect we'll see exceptions in the wild... :scream: